business/istio_validations.go

```diff
@@ -58,19 +58,17 @@ func (in *IstioValidationsService) GetValidations(namespace, service string) (mo
 	var services models.ServiceList
 	var namespaces models.Namespaces
 	var workloadsPerNamespace map[string]models.WorkloadList
-	var gatewaysPerNamespace [][]networking_v1alpha3.Gateway
 	var mtlsDetails kubernetes.MTLSDetails
 	var rbacDetails kubernetes.RBACDetails
 	var registryServices []*kubernetes.RegistryService
 
-	wg.Add(9) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
+	wg.Add(8) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
 
 	// We fetch without target service as some validations will require full-namespace details
 	go in.fetchIstioConfigList(&istioConfigList, namespace, errChan, &wg)
 	go in.fetchExportedResources(&exportedResources, namespace, errChan, &wg)
 	go in.fetchNamespaces(&namespaces, errChan, &wg)
 	go in.fetchAllWorkloads(&workloadsPerNamespace, errChan, &wg)
-	go in.fetchGatewaysPerNamespace(&gatewaysPerNamespace, errChan, &wg)
 	go in.fetchNonLocalmTLSConfigs(&mtlsDetails, namespace, errChan, &wg)
```

Comment:
A minor comment here to confirm that the scope of this work would be just the IstioConfigList query, isn't it ?
I see more improvements here but I see it would be better to do it in an iterative way.

Reply:
You are right.
There are 2 things combined in this PR, one is the IstioConfigList query, and the second is optimizing fetching Gateways per namespace and instead use the query result (which was the initial idea, but without exported query it would not be done), so I sent all together to save time in reviews.